lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1139,9 +1139,10 @@ FIRRTLModuleLowering::lowerModule(FModuleOp oldModule, Block *topLevelModule,
   if (auto svAttrs = sv::getSVAttributes(oldModule))
     sv::setSVAttributes(newModule, svAttrs);
 
-  // Pass along the number of random initialization bits needed for this module.
-  if (auto randomWidth = oldModule->getAttr("firrtl.random_init_width"))
-    newModule->setAttr("firrtl.random_init_width", randomWidth);
+  // Pass along FIRRTL dialect attributes.
+  for (auto attr : oldModule->getDiscardableAttrs())
+    if (isa_and_nonnull<FIRRTLDialect>(attr.getNameDialect()))
+      newModule->setAttr(attr.getName(), attr.getValue());
```

Comment:
If the goal is to enable external users to pass these annotations down to custom passes in HW, then I don't know if requiring that these are prefixed with `firrtl` is totally required. You can argue that any annotation should be passed onwards. I don't know if we want to force users of custom annotations to have to use the `firrtl.` prefix. If we don't know which ones are annotations then arguably the right thing to do is to pass on all discardable attributes.

Reply:
Annotation is highly coupled with FIRRTL (AnnotationSet for example) so I do want to require external users to lower annotations into something different before in LowerToHW. That said it's reasonable to me to pass through all discardable attributes regardless of their dialects.